--- gaol-4.2.0/Makefile.am	2016-11-02 18:05:57.775496711 +0100
+++ gaol-4.2.0/Makefile.am	2016-11-02 18:06:07.811889038 +0100
@@ -23,8 +23,8 @@
 #------------------------------------------------------------------------------
 
 EXTRA_DIST = BUGS msvc examples
-SUBDIRS = gdtoa gaol check gdtoa/test
-DIST_SUBDIRS = gdtoa gaol check doc gdtoa/test
+SUBDIRS = gdtoa gaol gdtoa/test
+DIST_SUBDIRS = gdtoa gaol doc gdtoa/test
 ACLOCAL_AMFLAGS = -I m4
 
 # Requires automake 1.4
--- gaol-4.2.0/configure.ac	2016-11-02 16:49:23.515135709 +0100
+++ gaol-4.2.0/configure.ac	2016-11-02 16:49:55.120391000 +0100
@@ -351,6 +351,11 @@
           [Define this if your system is a cygwin-based ix86 or compatible])
 	AM_CONDITIONAL([IS_CYGWIN_COND],[true])
 	;;
+*-*-msys*)
+	AC_DEFINE(IX86_MINGW,1,
+          [Define this if your system is a mingw-based ix86 or compatible])
+	AM_CONDITIONAL([IS_CYGWIN_COND],[false])
+	;;
 i?86-apple-*)
 	AC_DEFINE(IX86_MACOSX,1,
           [Define this if your system is a Linux-based ix86 or compatible])
--- gaol-4.2.0/gaol/gaol_version.h	2016-11-02 17:52:49.176757087 +0100
+++ gaol-4.2.0/gaol/gaol_version.h	2016-11-02 17:53:04.505358760 +0100
@@ -35,7 +35,7 @@
 # include "gaol/gaol_version_msvc.h"
 
 #elif defined (__MINGW32__)
-// Nothing for now. FIXME!
+# include "gaol/gaol_version_mingw.h"
 
 #elif defined (__GNUC__)
 # include "gaol/gaol_configuration.h"
--- gaol-4.2.0/gaol/Makefile.am	2016-11-03 09:14:08.096189553 +0100
+++ gaol-4.2.0/gaol/Makefile.am	2016-11-03 09:16:08.337464411 +0100
@@ -63,6 +63,7 @@
 	gaol_interval_parser.h gaol_config.h gaol_configuration.h gaol_expression.h gaol_limits.h \
 	gaol_expr_eval.h  gaol_expr_visitor.h gaol_parameters.h \
 	gaol_double_op_apmathlib.h gaol_double_op_crlibm.h gaol_flags.h gaol_parser.h gaol_double_op.h gaol_flt_output.h \
+	gaol_config_mingw.h gaol_version_mingw.h \
 	gaol_port.h gaol_fpu.h gaol_fpu_fenv.h gaol_profile.h gaol.h \
 	gaol_eval_stack.h gaol_init_cleanup.h gaol_version.h gaol_allocator.h \
 	gaol_interval_fpu.h gaol_interval_sse.h gaol_intervalf.h $(INTERVAL2_HEADER)
--- gaol-4.2.0/gdtoa/gdtoa.h	2016-11-02 17:41:31.922173915 +0100
+++ gaol-4.2.0/gdtoa/gdtoa.h	2016-11-02 17:42:24.568240369 +0100
@@ -117,7 +117,7 @@
 extern char* gdtoa ANSI((FPI *fpi, int be, ULong *bits, int *kindp,
 			int mode, int ndigits, int *decpt, char **rve));
 extern void freedtoa ANSI((char*));
-#if !IX86_MACOSX
+#if !IX86_MACOSX && !IX86_MINGW
 extern float  strtof ANSI((CONST char *, char **)) __GAOL_THROW();
 extern double strtod ANSI((CONST char *, char **)) __GAOL_THROW();
 #endif // IX86_MACOSX
