--- gaol-4.2.0/configure.ac	2016-11-02 16:49:23.515135709 +0100
+++ gaol-4.2.0/configure.ac	2016-11-02 16:49:55.120391000 +0100
@@ -346,7 +346,7 @@
 	AC_DEFINE(X86_64BITS,1,
 		  [Define this if the OS is a 64 bits one])
 	;;
-i?86-*-cygwin*)
+*-*-msys*)
 	AC_DEFINE(IX86_CYGWIN,1,
           [Define this if your system is a cygwin-based ix86 or compatible])
 	AM_CONDITIONAL([IS_CYGWIN_COND],[true])
